junit-platform-commons/junit-platform-commons.gradle.kts

```diff
@@ -11,3 +12,14 @@ javaLibrary {
 dependencies {
 	api("org.apiguardian:apiguardian-api:${Versions.apiGuardian}")
 }
+
+moditect {
+	addMainModuleInfo {
+		overwriteExistingFiles.set(true)
+		module {
+			moduleInfo {
+				name = "org." + project.name.replace('-', '.')
```

Comment:
I'm suprised there are no exports rules? Do you actually want to export all the packages?

Reply:
A draft is a draft is draft... _but only between Ferengi._

More to come here. Should look like https://github.com/junit-team/junit5/blob/3842a3459458fa6992f04f583cacf9618c58f47f/junit-platform-commons/src/main/java/module-info.java ... soon.

Reply:
What is a Ferengi :) All the qualified exports might be cumbersome, there's currently no shorter syntax for that in ModiTect (not even sure whether/how it could be done).

Reply:
Does ModiTect native `module-info.java` syntax? I'd really like to put those descriptors natively "on the disk" and let moditect consume them.

Reply:
Yes, you can do that. You can either reference an existing module-info.java file or put the equivalent content right into the build file using the `moduleInfoSource` element. You'll then not benefit from export wildcards and requires clauses derivation of course.

Reply:
Good to know and thanks for the heads-up.

- "no export wildcards" is fine with me, like to be explicit here.
- "no requires clauses derivation"  ... mh, don't want to lose that feature.

Is it possible to reference an existing `module-info.java` file and run a verification step from derivations made from project's configuration?

Reply:
No, that's not not supported currently. It's an interesting idea, though. OTOH, couldn't you simply compile with javac then? The compiler does that "validation" for you by failing when referencing a non-available module.

Reply:
> simply compile with javac

True. But that introduces other ... challenges. See those draft PRs for details:

- #1593 -- switch to Java 9 as the basis _(my favorite!!!111!!eleven!!!)_
- #1847 -- dedicated `module` source set

 Using ModiTect by-passes those (`javac`, Gradle, ...) configuration ... quirks, erm adjustments.